dist/clouduploader.rb

```diff
@@ -31,57 +97,83 @@ def self.credentials(data)
+      STDOUT.write("#{JSON.parse(result).inspect}\n")
+
+      run_command("az storage blob delete",
+                  "--container-name #{job_data['container']} --account-name #{job_data['storage_account']} -n #{remote_file_name} --debug",
+                  "Deleting")
+
+      run_command("az logout", "", "Logging out")
     end
-  end
 
-  def self.upload(platform, image_path, job_data_file, image_filename)
-    STDOUT.sync = true
-    start = Time.now
+    def self.uncompress(backend_image_file)
+      compressed_file = "#{File.join(Dir.pwd, backend_image_file)}.xz"
+      uncompressed_file = "#{File.join(Dir.pwd, backend_image_file)}.vhd"
+      run_command("cp", "#{backend_image_file} #{compressed_file}", "Copying the file '#{backend_image_file}' to working directory and rename to ensure uncompression")
+      run_command("xz", "-c -d #{compressed_file} > #{uncompressed_file}", "Uncompressing file '#{backend_image_file}'")
+      run_command("rm", "-f #{compressed_file}", "Cleaning up compressed file '#{compressed_file}'")
+      uncompressed_file
+    end
 
-    job_data = JSON.parse(File.read(job_data_file))
-    FileUtils.ln_s(image_path, File.join(Dir.pwd, image_filename)) # link file into working directory
+    def self.calculate_image_name(image_filename)
+      image_filename.gsub(/\.xz|\.vhdfixed|\.vhd/, '')
+    end
 
-    case platform
-    when 'ec2'
-      THIRTY_MINUTES = 1800
-      ENV['PYTHONUNBUFFERED'] = '1'
-      CloudUploader::EC2.upload(image_filename, job_data)
-    when 'azure'
-      # TODO: Add the Azure upload code here
-    else
-      abort('No valid platform. Valid platforms are ec2 and azure.')
+    def self.run_command(command, params, message, options = { logout_on_failure: true })
+      STDOUT.write("#{message}...")
+      out, err, status = Open3.capture3("#{command} #{params}")
+      if status.success?
+        STDOUT.write("[OK]\n")
+        out
+      else
+        STDOUT.write("[ERROR]\n\n")
+        run_command("az logout &", "", "Logging out", logout_on_failure: false) if options[:logout_on_failure]
+        STDOUT.write("\n\n---DEBUG INFO----------------------------------------------------\n" \
+                     "Running: '#{command}'\n\n" \
+                     "#{out}\n" \
+                     "---ERROR MESSAGE-------------------------------------------------\n\n")
+        abort(err)
+      end
     end
 
-    diff = Time.now - start
-    STDOUT.write("Upload took: #{Time.at(diff).utc.strftime("%H:%M:%S")}\n")
+    def self.uploaded_image_url(storage_account, container, remote_image)
+      "https://#{storage_account}.blob.core.windows.net/#{container}/#{remote_image}"
+    end
+
+    private_class_method :uploaded_image_url, :run_command, :calculate_image_name, :uncompress
   end
 end
 
 HOME = '/etc/obs/cloudupload'.freeze
-ENV['HOME'] = HOME
-raise 'Wrong number of arguments, please provide: user platform upload_file targetdata filename' unless ARGV.length == 5
+#ENV['HOME'] = HOME
```

Comment:
Please remove this if it's not needed

Reply:
Is needed, I left this by mistake sorry :smiley_cat: